zerver/actions/message_edit.py

```diff
@@ -1363,3 +1363,61 @@ def check_update_message(
         queue_json_publish("embed_links", event_data)
 
     return number_changed
+
+
+@transaction.atomic(durable=True)
+def check_resolve_topic(
+    user_profile: UserProfile,
+    message_id: int,
+    stream_id: Optional[int] = None,
+    topic_name: Optional[str] = None,
+) -> int:
+    """This will resolve a topic given a message id of the topic, and user profile.
+    It checks whether the user profile has the permission to resolve the topic
+    and raises a JsonableError if otherwise.
+    It returns the number changed.
+    """
+    message = access_message(user_profile, message_id, lock_message=True)
```

Comment:
This is also called in `is_resolve_topic_request`, I guess we should try removing one of them.

Reply:
updated!